restcomm/restcomm.extension.controller/src/main/java/org/restcomm/connect/extension/controller/ExtensionController.java

```diff
@@ -96,41 +106,79 @@ public ExtensionResponse executePreOutboundAction(final IExtensionRequest ier, L
                     logger.info( extension.getName()+" is enabled="+extension.isEnabled());
                 }
                 if (extension.isEnabled()) {
-                    response = extension.preOutboundAction(ier);
-                    //fail fast
-                    if (!response.isAllowed()){
-                        break;
+                    try {
+                        ExtensionResponse tempResponse = extension.preOutboundAction(ier);
+                        if (tempResponse != null) {
+                            response = tempResponse;
+                            //fail fast
+                            if (!tempResponse.isAllowed()) {
+                                break;
+                            }
+                        }
+                    } catch (Exception e) {
```

Comment:
do we want to catch just Exception? should we try to catch Throwable? im not sure, just wondering...
My perspective is Extensions are out of RC core, and RC core works as the platform for specific Extensions. Being the ExtensionController the border from RC to extension, it may make sense to catch Throwable instead,and provide a default recovery path..

Reply:
Changed to catch a Throwable.